src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/Service.cs

```diff
@@ -0,0 +1,399 @@
+
+            MlInformation result = new MlInformation(info);
+            return result;
+        }
+
+        static public MlInformationList GetModels(string name)
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            IntPtr info = IntPtr.Zero;
+            NNStreamerError ret = Interop.Service.GetAllModel(name, out info);
+            NNStreamer.CheckException(ret, "Failed to get the given models");
+
+            MlInformationList result = new MlInformationList(info);
+            return result;
+        }
+
+        static public void DeleteModel(string name, int version)
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            NNStreamerError ret = Interop.Service.DeleteModel(name, version);
+            NNStreamer.CheckException(ret, "Failed to delete the given model");
+        }
+
+        static public void AddResource(string name, string path, string description="")
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            if (string.IsNullOrEmpty(path))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property path is invalid");
+
+            NNStreamerError ret = Interop.Service.AddResource(name, path, description);
+            NNStreamer.CheckException(ret, "Failed to add resource");
+        }
+
+        static public void DeleteResource(string name)
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            NNStreamerError ret = Interop.Service.DeleteResource(name);
+            NNStreamer.CheckException(ret, "Failed to delete resource");
+        }
+
+        static public MlInformationList GetResource(string name)
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            if (string.IsNullOrEmpty(name))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "The property name is invalid");
+
+            IntPtr infoList = IntPtr.Zero;
+            NNStreamerError ret = Interop.Service.GetResource(name, out infoList);
+            NNStreamer.CheckException(ret, "Failed to get resource");
+
+            return new MlInformationList(infoList);
+        }
+    }
+
+    public class MlInformation {
```

Comment:
The keyword `Information` was already in use, so I changed it to `MlInformation`.
If you have any suggestions, I would be grateful.

Reply:
consider separating MlInformation and MlInformationList class.
In the directory Tizen.MachineLearning.Inference, MLInformation.cs and MLInformationList.cs

Reply:
Separated them
Thank you :)